src/testplan/testplanTestCollectionParser.ts

```diff
@@ -0,0 +1,45 @@
+import { TestInfo, TestSuiteInfo } from 'vscode-test-adapter-api';
+
+enum TestObjectType{APP, SUITE, TEST}
+
+export function parseTestSuites(content: string): (TestSuiteInfo | TestInfo)[] {
+
+    const suites: (TestSuiteInfo | TestInfo)[] = [];
+    const parentStack: (TestSuiteInfo | TestInfo)[] = [];
+    content.split(/[\r\n]+/)
+        .map(line => line.trim())
+        .filter(line => line)
+        .map(line => line!)
+        .forEach(line => {
+            const data = line.split('::');
+            const testRank = data.length - 1;
+            if (testRank === TestObjectType.TEST)
```

Comment:
How is this working? 😅 testRank is a number right?

Reply:
Yeah, this is where my missing typescript knowledge shined :D It probably can be done with maps and reduces and recursive calls.
An example result of test discovery:
`Primary
  Primary::AlphaSuite
    Primary::AlphaSuite::test_equality_passing
    Primary::AlphaSuite::test_equality_failing
    Primary::AlphaSuite::test_membership_passing
    Primary::AlphaSuite::test_membership_failing
    Primary::AlphaSuite::test_regex_passing
    Primary::AlphaSuite::test_regex_failing
Secondary
  Secondary::BetaSuite
    Secondary::BetaSuite::passing_testcase_one
    Secondary::BetaSuite::passing_testcase_two
`
Basically it iterates through the lines, testRank is the count of '::' per lines.
if testRank is
- 0, thats a App
- 1, thats a suite
- 2, thats a testcase
The order of the lines guarantee that every suite comes before its test, so what I do is store the last known app and suite and append the currently found test as its children